### DISCUSSION TITLE
How can I access provider in component

### DISCUSSION THREAD

**Discussion by AmanRaj1608** [rainbow-me/rainbowkit#568] [Q&A] [answered]
I have done react setup with the RainbowKitProvider. 

Now after I connect to my wallet I want to make a new (web3 / ethers) instance from the provider to call my smart contract. But I am unable to get the provider or NetworkId data. Also didn't find this in any examples and docs. Can someone help, please?

**Comment by saihaj** [comment, marked as answer]
you can use either get `account?.connector.getProvider()` from [`useAccount`](https://wagmi.sh/docs/hooks/useAccount) or just use the [`useProvider`](https://wagmi.sh/docs/hooks/useProvider)

**Comment by AmanRaj1608** [comment]
Yup found it!

```ts
import { useProvider } from "wagmi";

const provider = useProvider();
```
